### PR TITLE
Times starting at 00:00 caused infinite loop

### PR DESCRIPTION
**Pull request by joshuadegier**
As mentioned in https://github.com/spatie/opening-hours/issues/44, having opening hours starting at 00:00 caused an infinite loop (only in the occasion where you'd *only* have opening hours starting with 00:00).

This PR fixed the bug by checking if the `$timeOffRange->start()` is equal to `$time` besides checking wether or not a `$timeOffRange` contains the time. The PR is backed up with a test. All tests run fine with this addition.

### CODE REVIEW

File: src/OpeningHoursForDay.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ protected function findNextOpenInFreeTime(Time $time, TimeRange $timeRange, Time
             TimeRange::fromString($prevTimeRange->end().'-'.$timeRange->start()) :
             TimeRange::fromString('00:00-'.$timeRange->start());
 
-        if ($timeOffRange->containsTime($time)) {
+        if ($timeOffRange->containsTime($time) || $timeOffRange->start() == $time) {
```
Comment (freekmurze) [inline src/OpeningHoursForDay.php]: Use strict comparison here (`===`)

File: tests/OpeningHoursTest.php [hunk 2]
```diff
@@ -298,4 +298,15 @@ public function it_can_retrieve_a_list_of_exceptional_closing_dates()
...
+
+    /** @test */
+    public function it_works_when_starting_at_midnight()
+    {
+        $openingHours = OpeningHours::create([
+            'monday' => ['00:00-16:00'],
+        ]);
+
+        $nextTimeOpen = $openingHours->nextOpen(new DateTime());
+        $this->assertInstanceOf('\DateTime', $nextTimeOpen);
```
Comment (freekmurze) [inline tests/OpeningHoursTest.php]: Import that `DateTime` namespace

**Comment by joshuadegier** [comment]
@freekmurze I'm not sure why, by TravisCI doesn't run the phpunit tests (I had to use ./vendor/bin/phpunit locally aswel instead of the global phpunit).

File: src/OpeningHoursForDay.php (hunk 1 above)
Comment (joshuadegier) [reply src/OpeningHoursForDay.php]: Tests fail when using strict comparisation

**Comment by joshuadegier** [comment]
Think I've got it covered now, instead of comparing `$timeOffRange->start()` with `$time`, I've used the `isSame()` method like so:

```php
        if ($timeOffRange->containsTime($time) || $timeOffRange->start()->isSame($time)) {
            return $timeRange->start();
        }
```

**Comment by freekmurze** [comment]
Thanks again! 🙌

**Comment by joshuadegier** [comment]
Great! My pleasure.

**Pull request merged**